src/Component.tsx

```diff
@@ -0,0 +1,37 @@
+import { Component } from 'react';
+
+export class KeyPressed extends Component {
+  state = {
+    keyPressed: '',
+  };
+
+  componentDidMount() {
+    document.addEventListener('keyup',
+      (event: KeyboardEvent) => this.keyBoardHandler(event));
+  }
+
+  componentDidUpdate() {
+    document.addEventListener('keyup',
+      (event: KeyboardEvent) => this.keyBoardHandler(event));
+  }
```

Comment:
you don't need to set event listener in `componentDidUpdate`. it will work if you set it only in `componentDidMount`
```suggestion
  componentDidMount() {
    document.addEventListener('keyup',
      (event: KeyboardEvent) => this.keyBoardHandler(event));
  }
```